tools/firtool/firtool.cpp

```diff
@@ -819,11 +839,21 @@ static LogicalResult executeFirtool(MLIRContext &context) {
   applyDefaultTimingManagerCLOptions(tm);
   auto ts = tm.getRootScope();
 
+  // Set up the input file.
+  std::string errorMessage;
+  auto input = openInputFile(inputFilename, &errorMessage);
+  if (!input) {
+    llvm::errs() << errorMessage << "\n";
+    return failure();
+  }
+
   // Figure out the input format if unspecified.
   if (inputFormat == InputUnspecified) {
     if (StringRef(inputFilename).endswith(".fir"))
       inputFormat = InputFIRFile;
-    else if (StringRef(inputFilename).endswith(".mlir"))
+    else if (StringRef(inputFilename).endswith(".mlir") ||
+             StringRef(inputFilename).endswith(".bc") ||
```

Comment:
I think mlir bitcode uses ".mlirbc" suffix (e.g. `https://github.com/llvm/llvm-project/blob/main/mlir/test/Bytecode/invalid/invalid-ir_section-attr.mlirbc`) so it would be nice to adjust here.

Reply:
Good catch, thanks!